### ISSUE TITLE
Support text transformation parameters with `%dateformat()%`

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/SmartTemplate4#317]
It may advantageous in some cases to transform strings (such as day or month names) returned by `%dateformat()%`. This could be achieved by adding **one** optional textformat parameter such as:

```
%dateformat("e B Y",lowercase)% 
%dateformat("e B Y",uppercase)% 
%dateformat("e B Y",camelcase)% 
%dateformat("e B Y",capitalize)% 
```

It should also be possible to combine this with the `clipboard` parameter, e.g.:

```
%dateformat("e B Y",toclipboard,lowercase)% 
```

**Comment by Houdoe** [comment]
Correct spelling in Dutch:
Since 1954: lower case Until the middle of the 20th century, the names of months, days of the week and public holidays were written in upper case: January, Monday, Friday, etc. **The official glossary of 1954** stipulated that the names of months and days of the week should be written in lower case, as they are not unique names. This has not changed to this day. Only public holidays are capitalised: Easter, Good Friday, Maundy Thursday, etc.

from: https://onzetaal.nl/taalloket/oktober-october#:~:text=Sinds%201954%3A%20kleine%20letter&text=De%20offici%C3%ABle%20woordenlijst%20van%201954%20bepaalde%20dat%20namen%20van%20maanden,Vrijdag%2C%20Witte%20Donderdag%2C%20enz.

**Comment by RealRaven2000** [comment]
> Correct spelling in Dutch:
> Since 1954: lower case Until the middle of the 20th century, the names of months, days of the week and public holidays were written in upper case: January, Monday, Friday, etc. **The official glossary of 1954** stipulated that the names of months and days of the week should be written in lower case, as they are not unique names. This has not changed to this day. Only public holidays are capitalised: Easter, Good Friday, Maundy Thursday, et

this change request should probably go into its own issue number, then I can patch it much faster. 

In the meantime, you can use the new feature I made here:

[smartTemplate-fx-4.7.1pre80.zip](https://github.com/user-attachments/files/16652111/smartTemplate-fx-4.7.1pre80.zip)

this supports the text transformation parameters as described above. So you have a workaround immediately until the dutch locales are fixed and tested.

---

To try the version above download zip file and drag the file into Thunderbird Add-ons Manager (do not extract contents, it won't install like that)

**Comment by RealRaven2000** [comment]
Implemented in v4.8 published on 03/09/2024
